### ISSUE TITLE
Overflow in table of terminated processes

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#1849]
This currently happens in the master branch. @ChrisJefferson is working on it already, I am recording it to remember. This is needed for GAP 4.9 beta, as it fails `make testbugfix' test.
```
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-master-test/GAPCOPT\
S/64build/GAPTARGET/bugfix/label/kovacs/GAP-master-snapshot/tst/testbugfix/201\
7-10-20-runprocess.tst:44
# Input is:
 for i in [1..1000] do func2(); od;
# Expected output:
# But found:
#E Overflow in table of terminated processes
#E Overflow in table of terminated processes
#E Overflow in table of terminated processes
#E Overflow in table of terminated processes
...
```

**Comment by ChrisJefferson** [comment]
Plan for fixing this bug (it's going to be in multiple piecesc)

1) Add a hook into GAP so libraries can report children which have died ( https://github.com/gap-system/gap/pull/1851 , ready for merging )

2) Use that hook in IO ( https://github.com/gap-packages/io/pull/55  - proof of principle, needs documentation and tests )

**Comment by alex-konovalov** [comment]
For now, @ChrisJefferson in #1910 makes the process test pass by disabling the IO signal handler. This would have to be revisited after the next IO release.

**Comment by alex-konovalov** [comment]
Fixed now in #1851 in combination with IO 4.5.0 release. Thanks @ChrisJefferson @fingolfin
